sdk/core/core-tracing/lib/tracerProxy.ts

```diff
@@ -23,5 +28,8 @@ export function getTracer() {
   if (!cache.tracer) {
     cache.tracer = new NoOpTracer();
   }
+  if (env.AZURE_TRACING_DISABLED) {
+    cache.tracer = new NoOpTracer();
```

Comment:
`getTracer` gets called a lot, but right now you'd create a new no-op tracer each time... feels like we should maybe just have a static noop tracer that you can return early with instead of making a new one